dbms/src/Functions/array/arrayCompact.cpp

```diff
@@ -0,0 +1,107 @@
+#include <DataTypes/DataTypesNumber.h>
+#include <Columns/ColumnsNumber.h>
+#include "FunctionArrayMapped.h"
+#include <Functions/FunctionFactory.h>
+
+namespace DB
+{
+/// arrayCompact(['a', 'a', 'b', 'b', 'a']) = ['a', 'b', 'a'] - compact arrays
+    namespace ErrorCodes
+    {
+        extern const int ILLEGAL_COLUMN;
+    }
+
+    struct ArrayCompactImpl
+    {
+        static bool useDefaultImplementationForConstants() { return true; }
+        static bool needBoolean() { return false; }
+        static bool needExpression() { return false; }
+        static bool needOneArray() { return false; }
+
+        static DataTypePtr getReturnType(const DataTypePtr & expression_return, const DataTypePtr & /*array_element*/)
```

Comment:
The implementation of this function is definitely wrong (it has been copy-pasted from some other function).
The function arrayCompact simply returns the same type.

Reply:
That's my oversight. It has been corrected.